contracts/SwapRouter.sol

```diff
@@ -179,6 +179,13 @@ contract SwapRouter is
                     : sqrtPriceLimitX96,
                 abi.encode(data)
             );
+
+        uint256 amountOutReceived;
+        (amountIn, amountOutReceived) = zeroForOne
+            ? (uint256(amount0Delta), uint256(-amount1Delta))
+            : (uint256(amount1Delta), uint256(-amount0Delta));
+        // it's technically possible to not receive the full output amount
+        require(amountOutReceived == amountOut);
```

Comment:
hmm actually this breaks exactOutputSingle with meaningful price limits but i guess that's ok

Reply:
oh i didn't even think of that to be honest...odd, but i guess they can just use exactInputSingle? is this an argument to remove the price limit arg from exactOutputSingle?

Reply:
addressed by only running the check when price limit isn't specified